test/functional/rpc_getblockfrompeer.py

```diff
@@ -75,12 +80,42 @@ def run_test(self):
 
         self.log.info("Successful fetch")
         result = self.nodes[0].getblockfrompeer(short_tip, peer_0_peer_1_id)
-        self.wait_until(lambda: self.check_for_block(short_tip), timeout=1)
+        self.wait_until(lambda: self.check_for_block(node=0, hash=short_tip), timeout=1)
         assert_equal(result, {})
 
         self.log.info("Don't fetch blocks we already have")
         assert_raises_rpc_error(-1, "Block already downloaded", self.nodes[0].getblockfrompeer, short_tip, peer_0_peer_1_id)
 
+        self.log.info("Connect pruned node")
+        # We need to generate more blocks to be able to prune
+        self.connect_nodes(0, 2)
+        self.generate(self.nodes[0], 400)
+        pruneheight = self.nodes[2].pruneblockchain(300)
```

Comment:
If you have to retouch, if you call this `pruned_node = self.nodes[2]` the rest of the test is a bit easier to read.

Reply:
done